tests/workspace_smoke/doc_demo_smoke.py

```
@@ -0,0 +1,898 @@
+        log_file.flush()
+        if up_result.returncode != 0:
+            return Result(
+                demo.workspace,
+                demo.kind,
+                "failed",
+                demo.command,
+                demo.doc,
+                log_path,
+                reason="docker compose up failed",
+            )
+        process = subprocess.Popen(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)